spot_driver/include/spot_driver/conversions/robot_state.hpp

```diff
@@ -114,14 +116,15 @@ std::optional<tf2_msgs::msg::TFMessage> getTf(const ::bosdyn::api::RobotState& r
  * @param prefix  The prefix to apply to all robot frame IDs. This corresponds to the name of the robot. It is expected
  * to terminate with `/`.
  * @param preferred_base_frame_id Frame ID to use as the base frame of the TF tree. Must be either "odom" or "vision".
+ * @param frames_to_ignore Set of frames to not include in the TF tree.
  * @return If the input frame tree snapshot contains a non-zero number of entries, return a TFMessage containing
  * this data. Otherwise, return nullopt.
  */
 std::optional<tf2_msgs::msg::TFMessage> getTf(const ::bosdyn::api::FrameTreeSnapshot& frame_tree_snapshot,
                                               const google::protobuf::Timestamp& timestamp_robot,
                                               const google::protobuf::Duration& clock_skew, const std::string& prefix,
-                                              const std::string& preferred_base_frame_id);
-
+                                              const std::string& preferred_base_frame_id,
+                                              const std::set<std::string, std::less<>> frames_to_ignore = {});
```

Comment:
Do you need the `std::less<>`?

Reply:
I couldn't get the code to build without it unfortunately 
```
/home/khughes/spot_ros2_ws/src/spot_ros2/spot_driver/src/object_sync/object_synchronizer.cpp:509:34: error: no matching function for call to ‘getTf(const bosdyn::api::FrameTreeSnapshot&, const google::protobuf::Timestamp&, const google::protobuf::Duration&, std::string&, std::string&, const std::set<std::__cxx11::basic_string<char>, std::less<void> >&)’
...
/home/khughes/spot_ros2_ws/src/spot_ros2/spot_driver/include/spot_driver/conversions/robot_state.hpp:127:75: note:   no known conversion for argument 6 from ‘set<[...],less<void>>’ to ‘set<[...],less<std::__cxx11::basic_string<char>>>’
  127 |                                               const std::set<std::string> frames_to_ignore = {});

```

Reply:
Also, should it be a reference?